packages/app/routes/api/comments.js

```diff
@@ -0,0 +1,205 @@
+const express = require('express'),
+  router = new express.Router(),
+  Comment = require('../../models/Comment'),
+  User = require('../../models/User'),
+  { createNewComment } = require('../../lib/comments'),
+  passport = require('passport'),
+  log = require('../../config/log'),
+  { sendError } = require('../../lib/requestUtils'),
+  { logProjectComment } = require('../../lib/userActivity')
+const userFieldSelection = 'name email avatar company' // select only name and email from users
+const populateParams = [
+  {
+    path: 'user',
+    select: userFieldSelection
+  },
+  {
+    path: 'mentions',
+    select: userFieldSelection
+  },
+  {
+    path: 'replies',
+    populate: [
+      {
+        path: 'user',
+        select: userFieldSelection
+      },
+      {
+        path: 'mentions',
+        select: userFieldSelection
+      },
+      {
+        path: 'replies'
+      },
+      {
+        path: 'likes',
+        select: userFieldSelection
+      }
+    ]
+  },
+  {
+    path: 'likes',
+    select: userFieldSelection
+  }
+]
+
+router.get(
+  '/:commentId',
+  passport.authenticate('jwt', { session: false }),
+  async (req, res) => {
+    log.info(
+      {
+        requestId: req.id,
+        user: req.user.id,
+        comment: req.params.commentId
+      },
+      'User is getting comment replies'
+    )
+    try {
+      const comment = await Comment.findOne({
+        _id: req.params.commentId
+      }).populate(populateParams)
+      let { replies } = comment
+
+      if (req.query.sort === 'asc') {
+        replies = replies.sort((a, b) =>
+          a.likes.length > b.likes.length
+            ? 1
+            : b.likes.length > a.likes.length
+            ? -1
```

Comment:
I did not get it. Why are you verifying the opposite of "a.likes.length" is greater than "b.likes.length"? if you want to exclude the equals possibility you could use >= or <=

Reply:
You could use "a.likes.length - b.likes.length" as well instead of this logic.